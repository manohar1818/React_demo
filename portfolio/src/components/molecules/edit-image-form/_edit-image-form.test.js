import React from "react";
import { create } from "react-test-renderer";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import EditImageForm from "./_edit-image-form";
const name = "image1";
const nameError = "";
describe("EditImageForm component", () => {
  test("Matches the snapshot", () => {
    const editImageForm = create(
      <EditImageForm name={name} nameError={nameError} />
    );
    expect(editImageForm.toJSON()).toMatchSnapshot();
  });
  test("Matches the snapshot for name Error", () => {
    const editImageForm = create(
      <EditImageForm name={name} nameError="invalid name" />
    );
    expect(editImageForm.toJSON()).toMatchSnapshot();
  });
  test("component is rendered", () => {
    const { getByTestId, getByText } = render(
      <EditImageForm name={name} nameError={nameError} />
    );
    expect(getByTestId("editImageForm")).toBeInTheDocument();
  });

 
});
