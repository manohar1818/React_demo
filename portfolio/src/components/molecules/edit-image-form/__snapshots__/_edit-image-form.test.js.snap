// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditImageForm component Matches the snapshot 1`] = `
<form
  className="makeStyles-form-1"
  data-testid="editImageForm"
>
  <div
    className="makeStyles-styleInputFields-4"
    data-testid="inputField"
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline"
      data-testid="inputAtom"
      onClick={[Function]}
      required={true}
      style={
        Object {
          "color": "#000",
        }
      }
    >
      <input
        className="MuiInputBase-input MuiInput-input"
        data-testid="image1"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
      />
    </div>
  </div>
  <p
    className="MuiTypography-root makeStyles-error-3 MuiTypography-body1"
  />
  <div
    className="makeStyles-button-5"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      data-testid="button"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <span
          className="MuiTypography-root MuiTypography-button"
        >
           
          save
        </span>
      </span>
    </button>
  </div>
</form>
`;

exports[`EditImageForm component Matches the snapshot for name Error 1`] = `
<form
  className="makeStyles-form-1"
  data-testid="editImageForm"
>
  <div
    className="makeStyles-styleInputFields-4"
    data-testid="inputField"
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline"
      data-testid="inputAtom"
      onClick={[Function]}
      required={true}
      style={
        Object {
          "color": "#000",
        }
      }
    >
      <input
        className="MuiInputBase-input MuiInput-input"
        data-testid="image1"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
      />
    </div>
  </div>
  <p
    className="MuiTypography-root makeStyles-error-3 MuiTypography-body1"
  >
    invalid name
  </p>
  <div
    className="makeStyles-button-5"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
      data-testid="button"
      disabled={true}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <span
          className="MuiTypography-root MuiTypography-button"
        >
           
          save
        </span>
      </span>
    </button>
  </div>
</form>
`;
